gr-utils/modtool/core/disable.py

```diff
@@ -100,43 +100,45 @@ def _handle_cc_qa(cmake, fname):
         for subdir in self._subdirs:
             if self.skip_subdirs[subdir]:
                 continue
-            if self.info['version'] in ('37', '38') and subdir == 'include':
-                subdir = f'include/{self.info["modname"]}'
+            if self.info.version in ('37', '38') and subdir == 'include':
+                subdir = f'include/{self.info.modname}'
             try:
                 cmake = CMakeFileEditor(os.path.join(subdir, 'CMakeLists.txt'))
             except IOError:
                 continue
             logger.info(f"Traversing {subdir}...")
             filenames = []
-            if self.info['blockname']:
+            if self.info.blockname:
                 if subdir == 'python':
-                    blockname_pattern = f"^(qa_)?{self.info['blockname']}.py$"
+                    blockname_pattern = f"^(qa_)?{self.info.blockname}.py$"
                 elif subdir == 'python/bindings':
-                    blockname_pattern = f"^{self.info['blockname']}_python.cc$"
+                    blockname_pattern = f"^{self.info.blockname}_python.cc$"
                 elif subdir == 'python/bindings/docstrings':
-                    blockname_pattern = f"^{self.info['blockname']}_pydoc_template.h$"
+                    blockname_pattern = f"^{self.info.blockname}_pydoc_template.h$"
                 elif subdir == 'lib':
-                    blockname_pattern = f"^{self.info['blockname']}_impl(\\.h|\\.cc)$"
-                elif subdir == self.info['includedir']:
-                    blockname_pattern = f"^{self.info['blockname']}.h$"
+                    blockname_pattern = f"^{self.info.blockname}_impl(\\.h|\\.cc)$"
+                elif subdir == self.info.includedir:
+                    blockname_pattern = f"^{self.info.blockname}.h$"
                 elif subdir == 'grc':
-                    blockname_pattern = f"^{self.info['modname']}_{self.info['blockname']}.block.yml$"
+                    blockname_pattern = f"^{self.info.modname}_{self.info.blockname}.block.yml$"
                 if blockname_pattern:
                     filenames = cmake.find_filenames_match(blockname_pattern)
-            elif self.info['pattern']:
-                filenames = cmake.find_filenames_match(self.info['pattern'])
-            yes = self.info['yes']
+            elif self.info.pattern:
+                filenames = cmake.find_filenames_match(self.info.pattern)
+            yes = (not self.info.no) and self.info.yes
```

Comment:
Same here, I think ToolConfig could catch invalid --no --yes settings, and make sure that `info.no` and `info.yes` are valid.